service/history/historybuilder/event_factory.go

```diff
@@ -395,13 +395,19 @@ func (b *EventFactory) CreateWorkflowExecutionTerminatedEvent(
 
 func (b *EventFactory) CreateWorkflowExecutionOptionsUpdatedEvent(
 	versioningOverride *workflowpb.VersioningOverride,
+	attachRequestID string,
+	attachCompletionCallbacks []*commonpb.Callback,
+	links []*commonpb.Link,
```

Comment:
Maybe instead accept a historypb.WorkflowExecutionOptionsUpdatedEventAttributes and links here to avoid this getting out of control?

Reply:
Hm, I think it would make sense to use `historyservice.UpdateWorkflowExecutionOptionsRequest`, but it would add some logic in the event factory (do the mask processing here), which I'm not sure it makes sense.
I'd leave as it is for now, and refactor later. I'm already planning to refactor the `startworkflow` api to call the `updateworkflowwithoptions` api in which I can address this issue together.